include/fastrtps/transport/UDPChannelResource.h

```diff
@@ -132,12 +136,54 @@ class UDPChannelResource : public ChannelResource
         return message_receiver_;
     }
 
+    inline virtual void disable() override
+    {
+        ChannelResource::disable();
+    }
+
+    inline bool closing()
+    {
+        return closing_.load();
+    }
+
+    void release(
+            const Locator_t& locator,
+            const asio::ip::address& address);
+
+protected:
+    /**
+     * Function to be called from a new thread, which takes cares of performing a blocking receive
+     * operation on the ReceiveResource
+     * @param input_locator - Locator that triggered the creation of the resource
+    */
+    void perform_listen_operation(
+            Locator_t input_locator);
+
+    /**
+    * Blocking Receive from the specified channel.
+    * @param receive_buffer vector with enough capacity (not size) to accomodate a full receive buffer. That
+    * capacity must not be less than the receive_buffer_size supplied to this class during construction.
+    * @param receive_buffer_capacity Maximum size of the receive_buffer.
+    * @param[out] receive_buffer_size Size of the received buffer.
+    * @param[out] remote_locator Locator describing the remote restination we received a packet from.
+    */
+    bool Receive(
+            octet* receive_buffer,
+            uint32_t receive_buffer_capacity,
+            uint32_t& receive_buffer_size,
+            Locator_t& remote_locator);
+
 private:
 
     TransportReceiverInterface* message_receiver_; //Associated Readers/Writers inside of MessageReceiver
     eProsimaUDPSocket socket_;
     bool only_multicast_purpose_;
     std::string interface_;
+    UDPTransportInterface* transport_;
+    std::atomic<bool> closing_;
```

Comment:
Needs to be atomic?

Reply:
Since it is protected in all places it is used, isn't needed to be atomic.

Reply:
I removed the public `closing()` method because wasn't being used and now it would require to be protected.